src/Common/FitOnFhir.Common/Services/AuthStateService.cs

```diff
@@ -81,5 +100,62 @@ public AuthState CreateAuthState(HttpRequest httpRequest)
 
             return new AuthState(ExternalIdentifier, ExternalSystem);
         }
+
+        public async Task<AuthState> RetrieveAuthState(string nonce, CancellationToken cancellationToken)
+        {
+            try
+            {
+                // Get a reference to the blob
+                BlobClient blobClient = _blobContainerClient.GetBlobClient(nonce);
+
+                // Get the AuthState
+                var response = await blobClient.DownloadAsync(cancellationToken);
+
+                // deserialize the AuthState
+                StreamReader reader = new StreamReader(response.Value.Content);
+                string json = reader.ReadToEnd();
+                return AuthState.Parse(json);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Failed to retrieve the auth state.");
+                throw;
+            }
+        }
+
+        public async Task<string> StoreAuthState(AuthState state, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var nonce = GenerateNonce(Constants.NonceLength);
+
+                // Get a reference to the blob
+                BlobClient blobClient = _blobContainerClient.GetBlobClient(nonce);
+
+                // Set the CacheControl property to expire in 5 minutes (300 seconds)
+                await blobClient.SetHttpHeadersAsync(new BlobHttpHeaders { CacheControl = "max-age=300" }, cancellationToken: cancellationToken);
```

Comment:
I don't think the CacheControl header is what you want to use. I believe this is a way for clients to expire local caches. This won't actually delete the blob after 5 minutes.

I think this is what you want:

https://docs.microsoft.com/en-us/rest/api/storageservices/set-blob-expiry#expiryoption

Reply:
This is what I had found, that led me to put that in:
https://docs.microsoft.com/en-us/azure/cdn/cdn-manage-expiration-of-blob-content?toc=%2Fazure%2Ffrontdoor%2FTOC.json

Reply:
That document is a very specific use case when a Content Distribution Network (CDN) is connected to Azure Blob Store. This provides a way for the CDN to cache the blob internally and serve it to consumers for a specified period of time without making a request to blob every time. Once the max-age is reached, the CDN will fetch the blob again and cache that version of the blob until the max-age expires the blob again.

In other words, the blob is never deleted, it just provides a way to control how often a consumer needs to update their cache.

Reply:
From the docs link you shared, note that it only applies to Hierarchical Namespace enabled accounts.  A quick check of our storage account resource created by the template reveals that Hierarchical Namespaces are disabled.  If I attempt to enable that, I am redirected to a page which talks about upgrading "to a storage account with Azure Data Lake Gen2 capabilities".

This is not looking very cut and dried.  I am also surprised and disappointed, that there is no direct C# method or property to do something as simple as set an expiration for a blob.

Reply:
What happens if you use the storage account as is and just set the `x-ms-expiry-option` header?